### ISSUE TITLE
Conflicting pull-requests accepted to fix postgres schema:log with uppercase table names and enums

### ISSUE THREAD

**Issue by romain-gilliotte** [typeorm/typeorm#3828]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[ ] `0.2.15` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

See https://github.com/typeorm/typeorm/issues/3536 and https://github.com/typeorm/typeorm/issues/3792

**What happened:**

Accepting both pull-request is causing typeorm migration tool to crash again when trying to generate migrations on databases created with any version (<= 0.2.14) that contained:
- uppercase letters in the table names.
- at least one enum in one of those tables.

```
{ QueryFailedError: type "transaction.detail_type_enum" does not exist
    at new QueryFailedError (.../src/error/QueryFailedError.ts:9:9)
    at Query.callback (.../src/driver/postgres/PostgresQueryRunner.ts:176:30)
    at Query.handleError (.../node_modules/pg/lib/query.js:142:17)
    at Connection.connectedErrorMessageHandler (.../node_modules/pg/lib/client.js:160:17)

# The enum is named "*D*etail_type_enum" in my database, because it was created by an older
# version of typeorm
```
When TypeOrm named the enums before this version, it kept uppercase letters from the table name when generating the enum type name, it no longer does.

My pull-request corrected the missing quotes in the driver that were causing the crash when uppercase letters existed in enums names.

@onionhammer pull-request corrected the same problem by keeping bad quoting, but lowercasing all enum names.

Accepting both patches into the 0.2.15 version cause typeorm to **both** lowercase and quote the enum type, which result in the enum type not being found, and crashing again.

Both solutions were OK but don't play well together

**Comment by romain-gilliotte** [comment]
Actually i wrote this issue a bit too fast.
I think that the PR that came with https://github.com/typeorm/typeorm/issues/3536 was only fixing the problem for database created with typeorm >= 0.2.15.

Even without quoting, postgres would not be able to find an enum type created with a former version of typeorm.

I'll work around the issue by creating a migration manually to rename the enums to lowercase.

**Comment by vlapo** [comment]
Man you are right. Your solution in https://github.com/typeorm/typeorm/commit/836e5a0cd95758eed5ee040521bc9ed52836cb31 resolve root problem of both issues #3536 and #3792. I did not see problem with quotes before but I tried run test for #3536 with your fix and it works. I wanted to fix small issue fast and did not see root cause.

So what now? Release new version with reverting lowercase enum. Or stay with this breaking change for migrations.

**Comment by romain-gilliotte** [comment]
My bad!
I should have commented issue #3536 instead of creating a new one.
I just did not find it when I looked for it.

Either way, to me the root problem here is that the migration generator is not reading the name of the enum from the database to see if it is different from what is wanted.

@see https://github.com/typeorm/typeorm/blob/master/src/driver/postgres/PostgresQueryRunner.ts#L581

It just assumes that the name of the enum is the one that would be generated with the running version of typeorm with the old column data.

**I did not try** so I'm possibly wrong (i am not sure how `changeColumn` is triggered), but reading the code, it's probable that typeorm can be crashed by doing the following:

1. Create model with enum (lower/uppercase should not matter).
2. Sync your database
3. Rename the `enum` type by hand (by running `ALTER TYPE my_enum RENAME TO my_orm_breaking_enum`)
4. Run either schema:sync or schema:log should crash on the same error (type not found) 

I'll try to write a test for this on the weekend if I get the time

**Comment by vlapo** [comment]
I created PR with revert changes from my previous PR. But I keep tests as they are so we have a prove #3793 fixes also #3536. :)

But I think your PR #3889 with tests will failing every time also after reverting. So I think we can close #3889.
